packages/webamp-modern-2/src/skin/makiClasses/Movable.ts

```diff
@@ -0,0 +1,203 @@
+/**
+ * this file is needed to workaround of button-moving-layout issue
+ */
+
+import { toBool } from "../../utils";
+import GuiObj from "./GuiObj";
+
+import Layout from "./Layout";
+import { LEFT, RIGHT, TOP, BOTTOM, CURSOR, MOVE } from "../Cursor";
+
+export default class Movable extends GuiObj {
+  _movable: boolean = false;
+  _resizable: number = 0;
+  _resize: string;
+  _resizingEventsRegisterd: boolean = false;
+  _movingEventsRegisterd: boolean = false;
+
+  setXmlAttr(_key: string, value: string): boolean {
+    const key = _key.toLowerCase();
+    if (super.setXmlAttr(key, value)) {
+      return true;
+    }
+    switch (key) {
+      case "move":
+        this._movable = toBool(value);
+        this._renderCssCursor();
+        break;
+      case "resize":
+        this._resize = value == "0" ? "" : value;
+        this._renderCssCursor();
+        break;
+
+      default:
+        return false;
+    }
+    return true;
+  }
+
+  _renderCssCursor() {
+    // this._resizable = 0; //flag. replace soon
+    this._div.style.cursor = "none"; //flag. replace soon
+    switch (this._resize) {
+      case "right":
+        this._div.style.cursor = "e-resize";
+        this._resizable = RIGHT;
+        break;
+      case "left":
+        this._div.style.cursor = "w-resize";
+        this._resizable = LEFT;
+        break;
+      case "top":
+        this._div.style.cursor = "n-resize";
+        this._resizable = TOP;
+        break;
+      case "bottom":
+        this._div.style.cursor = "s-resize";
+        this._resizable = BOTTOM;
+        break;
+      case "topleft":
+        this._div.style.cursor = "nw-resize";
+        this._resizable = TOP | LEFT;
+        break;
+      case "topright":
+        this._div.style.cursor = "ne-resize";
+        this._resizable = TOP | RIGHT;
+        break;
+      case "bottomleft":
+        this._div.style.cursor = "sw-resize";
+        this._resizable = BOTTOM | LEFT;
+        break;
+      case "bottomright":
+        this._div.style.cursor = "se-resize";
+        this._resizable = BOTTOM | RIGHT;
+        break;
+      default:
+        // this._div.style.removeProperty('cursor');
```

Comment:
We don't need to clear whatever the cursor style was before in this case?

Reply:
Yeah, I have a better idea, to revalidate both move & resize in one function. done.